pkg/runtime/conversion_generator.go

```diff
@@ -19,40 +19,69 @@ package runtime
 import (
 	"fmt"
 	"io"
+	"path"
 	"reflect"
 	"sort"
 	"strings"
 
 	"github.com/GoogleCloudPlatform/kubernetes/pkg/conversion"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
 )
 
 type ConversionGenerator interface {
 	GenerateConversionsForType(version string, reflection reflect.Type) error
 	WriteConversionFunctions(w io.Writer) error
-	RegisterConversionFunctions(w io.Writer) error
+	RegisterConversionFunctions(w io.Writer, pkg string) error
+	AddImport(pkg string) string
+	RepackImports(exclude util.StringSet)
+	WriteImports(w io.Writer) error
 	OverwritePackage(pkg, overwrite string)
+	AssumePrivateConversions()
```

Comment:
Right now conversions for Kube objects are private methods.  OpenShift embeds some of the Kube objects (pod, type meta, object meta) and so when the OpenShift private convertor functions they have to go back to scheme.Convert in order to get access to the method.  Ideally we'd either make the core conversions public or visible through a fast path named accessor, but this was the compromise I made for now.

> On Jun 19, 2015, at 4:08 AM, Wojciech Tyczynski <notifications@github.com> wrote:
> 
> In pkg/runtime/conversion_generator.go:
> 
> >  	OverwritePackage(pkg, overwrite string)
> > +	AssumePrivateConversions()
> What is the purpose of it? I don't see it used anywhere and I'm not sure if I understand why it is really needed.
> 
> —
> Reply to this email directly or view it on GitHub.
>